api/main.py

```diff
@@ -216,7 +216,13 @@ async def get_nodes(request: Request, kind: str = "node"):
                 detail=f"Invalid request parameters: {msg}"
         )
 
-    translated_params = model.translate_fields(query_params)
+    is_translated, translated_params = model.translate_fields(query_params)
+    if not is_translated:
```

Comment:
It seems like there's a lot of overhead with throwing exceptions, catching exceptions and returning status.  Maybe it would be good to review the whole error propagation and handling code.  Ideally, I think exceptions should only be caught in the endpoint implementation in `main.py` to convert them to HTTP responses.

Reply:
I think that can be a follow-up PR.

Reply:
OK

Reply:
Added a follow-up PR for this https://github.com/kernelci/kernelci-api/pull/200